SofaKernel/modules/Sofa.GL/Component/Rendering3D/src/sofa/gl/component/rendering3d/OglGrid.cpp

```diff
@@ -55,120 +67,133 @@ void OglGrid::reinit()
-             plane.getValue() == "X" ||
-            plane.getValue() == "zOy" ||
-            plane.getValue() == "ZOY" ||
-            plane.getValue() == "yOz" ||
-            plane.getValue() == "YOZ")
+    const auto planeValue = d_plane.getValue();
+
+    if (planeValue == "x" ||
+             planeValue == "X" ||
+            planeValue == "zOy" ||
+            planeValue == "ZOY" ||
+            planeValue == "yOz" ||
+            planeValue == "YOZ")
     {
         internalPlane = PLANE_X;
     }
-    else if (plane.getValue() == "y" ||
-             plane.getValue() == "Y" ||
-             plane.getValue() == "zOx" ||
-             plane.getValue() == "ZOX" ||
-             plane.getValue() == "xOz" ||
-             plane.getValue() == "XOZ")
+    else if (planeValue == "y" ||
+             planeValue == "Y" ||
+             planeValue == "zOx" ||
+             planeValue == "ZOX" ||
+             planeValue == "xOz" ||
+             planeValue == "XOZ")
     {
         internalPlane = PLANE_Y;
     }
-    else if (plane.getValue() == "z" ||
-             plane.getValue() == "Z" ||
-             plane.getValue() == "xOy" ||
-             plane.getValue() == "XOY" ||
-             plane.getValue() == "yOx" ||
-             plane.getValue() == "YOX")
+    else if (planeValue == "z" ||
+             planeValue == "Z" ||
+             planeValue == "xOy" ||
+             planeValue == "XOY" ||
+             planeValue == "yOx" ||
+             planeValue == "YOX")
     {
         internalPlane = PLANE_Z;
     }
     else
     {
-        msg_error() << "Plane parameter " << plane.getValue() << " not recognized. Set to z instead";
-        plane.setValue("z");
+        msg_error() << "Plane parameter " << d_plane.getValue() << " not recognized. Set to z instead";
+        d_plane.setValue("z");
         internalPlane = PLANE_Z;
     }
 
-    int nb = nbSubdiv.getValue();
+    const int nb = d_nbSubdiv.getValue();
     if (nb < 2)
     {
         msg_error() << "nbSubdiv should be > 2";
-        nbSubdiv.setValue(2);
+        d_nbSubdiv.setValue(2);
     }
 
-    //bounding box for the camera
-//    Real s = size.getValue();
-//    Coord min,max;
-//    switch(internalPlane)
-//    {
-//        case PLANE_X:
-//            min = Coord(-s*0.1, -s*0.5, -s*0.5);
-//            max = Coord(s*0.1, s*0.5, s*0.5);
-//            break;
-//        case PLANE_Y:
-//            min = Coord(-s*0.5, -s*0.1, -s*0.5);
-//            max = Coord(s*0.5, s*0.1, s*0.5);
-//            break;
-//        case PLANE_Z:
-//            min = Coord(-s*0.5, -s*0.5, -s*0.1);
-//            max = Coord(s*0.5, s*0.5, s*0.1);
-//            break;
-//    }
-//    f_bbox.setValue(sofa::type::BoundingBox(min,max));
+    buildGrid();
 
+    //bounding box for the camera
+    auto s = d_size.getValue();
+    sofa::type::Vec3f min,max;
+    switch(internalPlane)
+    {
+        case PLANE_X:
+            min = sofa::type::Vec3f(-s*0.1f, -s*0.5f, -s*0.5f);
```

Comment:
maybe create 2 variables for s*0.1f and s*0.5f , if you think we will change that values some days...